pkg/reconciler/v1alpha1/route/resources/virtual_service.go

```diff
@@ -149,12 +149,29 @@ func makeVirtualServiceRoute(domains []string, ns string, targets []traffic.Revi
 	route := v1alpha3.HTTPRoute{
 		Match:   matches,
 		Route:   weights,
-		Timeout: DefaultRouteTimeout,
+		Timeout: fmt.Sprintf("%ds", getRouteTimeout(targets)),
 	}
 	// Add traffic rules for activator.
 	return addActivatorRoutes(&route, ns, inactive)
 }
 
+func getRouteTimeout(targets []traffic.RevisionTarget) int {
+	maxTimeoutSeconds := 0
+	for _, t := range targets {
+		if t.Percent == 0 {
+			continue
+		}
+		targetTimeoutSeconds := t.TimeoutSeconds
+		if targetTimeoutSeconds == 0 {
+			targetTimeoutSeconds = DefaultRouteTimeoutSeconds
+		}
+		if maxTimeoutSeconds < targetTimeoutSeconds {
+			maxTimeoutSeconds = targetTimeoutSeconds
+		}
+	}
+	return maxTimeoutSeconds
```

Comment:
This doesn't fix #487 since it takes the max, right?  We should leave #487 open and consider implementing this in the `queue-proxy` for now.

Reply:
No, it doesn't fix #487. Agreed that we should leave it open. I updated the description of PR.